pkgs/build-support/fetchgit/nix-prefetch-git

```diff
@@ -148,7 +148,17 @@ checkout_hash(){
         hash=$(hash_from_ref "$ref")
     fi
 
-    clean_git fetch -t ${builder:+--progress} origin || return 1
+    if [[ -n "$deepClone" ]]; then
+        clean_git fetch -t ${builder:+--progress} origin || return 1
+        # Try to shallow fetch specific hash if it doesn't exist in any ref.
+        # For example, if a ref was force-pushed but the server still exposes the old hash.
+        clean_git fetch ${builder:+--progress} --depth 1 origin "$hash" 2> /dev/null
+    else
+        # Try to shallow fetch specific hash if server supports uploadpack.allowReachableSha1InWant.
+        # Fallback to full fetch on failure.
+        clean_git fetch ${builder:+--progress} --depth 1 origin "$hash" 2> /dev/null ||
+        clean_git fetch -t ${builder:+--progress} origin || return 1
```

Comment:
Are you missing a ``\`` here or is the first ``||`` to much? Also I think we want to add ``(...)``` here to make it clearer what happens if one fails and one succeeds.